leshan-tl-cf-shared-oscore/src/main/java/org/eclipse/leshan/transport/californium/oscore/cf/InMemoryOscoreContextDB.java

```diff
@@ -111,4 +118,24 @@ private static OSCoreCtx deriveContext(OscoreParameters oscoreParameters) {
             return null;
         }
     }
+
+    private static OSCoreCtx deriveClientContext(OscoreParameters oscoreParameters) {
+        try {
+            OSCoreCtx osCoreCtx = new OSCoreCtx(oscoreParameters.getMasterSecret(), true,
+                    oscoreParameters.getAeadAlgorithm(), oscoreParameters.getSenderId(),
+                    oscoreParameters.getRecipientId(), oscoreParameters.getHmacAlgorithm(), 32,
+                    oscoreParameters.getMasterSalt(), null, 1000);
+            osCoreCtx.setContextRederivationEnabled(true);
+            osCoreCtx.setContextRederivationPhase(PHASE.CLIENT_INITIATE);
+
+            return osCoreCtx;
+        } catch (OSException e) {
+            LOG.error("Unable to derive context from {}", oscoreParameters, e);
+            return null;
+        }
+    }
+
+    public void setClientRole(boolean clientRole) {
+        this.clientRole = clientRole;
+    }
```

Comment:
Using a setter means that we can change the mode after `InMemoryOscoreContextDB` is created.

I'm not sure we want to be able to do that, so maybe better to add a constructor argument instead of a setter ?
If you want you can add a second constructor and use a default clientRole = false for the already existing one.

Reply:
Yes, I see, that can make sense. I did it like that now in commit f480653b1f302be6ffbeab1bf4f27a5d0b026130